decompiler_agent/src/main/java/org/jrd/agent/api/Variables.java

```diff
@@ -0,0 +1,95 @@
+package org.jrd.agent.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Variables {
+
+    /**
+     * we have to have all inernal classes inited, so any later classes can use them without manual INIT.
+     */
+    public static void init() {
+        NoSuchFakeFieldException.init();
+        Global.init();
+    }
+
+    protected Variables() {
+        // prevents calls from subclass... checkstile sucks
```

Comment:
If you have a utility/static class such as this, it is good practice to have a private constructor. The exception throw is unnecessary AFAIK.

Reply:
hmm checstyle had survived t..ok